packages/e2e-tests/vite-ssr-esm/vite.config.js

```diff
@@ -14,6 +14,11 @@ export default defineConfig(({ command, mode }) => {
 				}
 			}
 		},
+		// TODO: investigate the condition issue. it's the same thing dominik and ben found.
+		// idk why it's only happening after https://github.com/vitejs/vite/pull/18395
+		ssr: {
+			noExternal: ['esm-env']
```

Comment:
if this is a general issue, should vite-plugin-svelte add that to the users config by default?!

Reply:
thats cheating the ecosystem-ci, users apps will run into this very issue then....

Reply:
so we definetely need to figure this out before supporting vite 6

Reply:
it could get lost in the storm, prefer to let this fail until we have a fix in vps. given svelte itself uses esm-env and sveltekit iirc also did the noExternal thing for it that should be an easy one. But i am also not convinced this fixes it in all projects unless users add the right conditions to their starting arguments.

Reply:
Removed 👍 Yeah given SvelteKit also did that, maybe we could also do that for now and call it a day. But I think the source of the problem still boils down to `esm-env` not working well with Vite without `--conditions` properly set, so we should look at this as not only a v-p-s problem.